src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReferenceShardTest.java

```diff
@@ -0,0 +1,67 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.genomics.dataflow.readers.bam.ReadConverter;
+import com.google.common.collect.Lists;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GoogleGenomicsReadToGATKReadAdapter;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public final class ReferenceShardTest {
+
+    @DataProvider(name = "reads")
+    public Object[][] reads(){
+        List<GATKRead> reads = Arrays.asList(makeRead(1, 300, 1), makeRead(100000, 10, 2), makeRead(299999, 2, 3));
+        List<ReferenceShard> referenceShards = Arrays.asList(new ReferenceShard(0, "1"), new ReferenceShard(1, "1"), new ReferenceShard(2, "1"));
+        return new Object[][]{
```

Comment:
This is a very unconventional Dataprovider setup.   You're putting all of the data into a single array and looping over the array within the test.  The purpose of the dataprovider is usually to avoid having to do the explicit looping in your test.

I would restructure like this:

```
    @DataProvider(name = "reads")
    public Object[][] reads(){
          return new Object[][]{
                {makeRead(1,300,1), new ReferenceShard(0,"1")},  //right in the middle of the shard
                {makeRead(100000, 10, 2), new ReferenceShard(1, "1")}, //at  the start of a shard
                {makeRead(299999, 2, 3),  new ReferenceShard(2, "1")}  //overlapping the end of a shard
        };
    }

    @Test(dataProvider = "reads")
    public void getVariantShardsFromIntervalTest(GATKRead r, ReferenceShard expectedShard) {
        ReferenceShard foundShard = ReferenceShard.getShardNumberFromInterval(r);
        Assert.assertEquals(foundShard, expectedShard);
    }
```

This will auto generate 3 tests, one for each array in the dataprovider.  It also lets you line up the inputs and expected outputs and label them in a nice easy to read grid.